python/hsfs/core/feature_group_engine.py

```diff
@@ -388,3 +390,65 @@ def save_feature_group_metadata(
                 feature_group_id=feature_group.id,
             )
         )
+
+    def list_feature_groups(
+        self,
+        latest_version_only: bool,
+        online_enabled_only: bool,
+        spine_only: bool,
+        external_only: bool,
+        with_features: bool,
+    ) -> List[fg.FeatureGroup, fg.ExternalFeatureGroup, fg.SpineGroup]:
+        fg_list = self._feature_group_api.get_all(
+            feature_store_id=self._feature_store_id,
+            with_features=with_features,
+        )
+
+        if online_enabled_only:
+            fg_list = [fgroup for fgroup in fg_list if fgroup.online_enabled]
+        if spine_only:
+            fg_list = [
+                fgroup for fgroup in fg_list if isinstance(fgroup, fg.SpineGroup)
+            ]
+        if external_only:
+            fg_list = [
+                fgroup
+                for fgroup in fg_list
+                if isinstance(fgroup, fg.ExternalFeatureGroup)
+            ]
+
+        if latest_version_only:
+            fg_list = [
+                fgroup
+                for fgroup in fg_list
+                if fgroup.version
+                == max(fg1.version for fg1 in fg_list if fg1.name == fgroup.name)
```

Comment:
may be using filter() is better for readability instead of having a for loop again